sisl/io/sile.py

```diff
@@ -306,10 +308,12 @@ def get_siles(attrs=None):
         return list(__siles)
 
     siles = []
-    for sile in __siles:
+    objs = __sile_rules if rules else __siles
```

Comment:
I don't like these one liners, ;)
You seem to use them a lot but I think it is better to do it explicitly.

Also, could you create a new method, `get_sile_rules`, it seems more appropriate?

Reply:
Hahahah yes sorry. It is because I'm writing pyhton and javascript at the same time and in javascript is quite common.